### DISCUSSION TITLE
Serialization safety

### DISCUSSION THREAD

**Discussion by skaarj1989** [jrouwe/JoltPhysics#921] [Q&A] [answered]
Is there a reason why serialization does not use tags/magic numbers?
Without that, there is a chance for a nasty error.

![image](https://github.com/jrouwe/JoltPhysics/assets/34756939/dda23c4f-41ba-45af-8e7e-42ea13ac6b16)

**Comment by jrouwe** [comment, marked as answer]
The idea is that these functions are used as part of your own game data serialization code, so they do not include any version numbering as versioning is expected to be handled by the game that is using the library. If you update Jolt then you have to ensure that whatever data you cooked is recooked again.

See: https://jrouwe.github.io/JoltPhysics/index.html#saving-shapes

**Reply by skaarj1989** [reply]
Thanks for the clarification.

**Comment by jrouwe** [comment]
B.t.w. the main reason for this is speed: If the library needs to be able to load every previous version then it will be filled with patching code. We need the data to stream in with little CPU intervention.
